### PR TITLE
IBP-5924 granular permissions to MG Studies

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review

Included:
https://github.com/IntegratedBreedingPlatform/BMSAPI/pull/809
https://github.com/IntegratedBreedingPlatform/Workbench/pull/829
https://github.com/IntegratedBreedingPlatform/Fieldbook/pull/1164

### CODE REVIEW

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 1]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-71">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_CROSSING_OPTIONS';
+             </sqlCheck>
+ 		</preConditions>
+        <comment>Crossing Options permission</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1394]: Use Add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 2]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-72">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_EXPORT_CROSSING_TEMPLATE';
+             </sqlCheck>
+ 		</preConditions>
+        <comment>Export Crossing Template permission</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1414]: Use Add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 3]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-73">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_IMPORT_CROSSES';
+             </sqlCheck>
+ 		</preConditions>
+        <comment>Import Crosses permission</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1434]: Use add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 4]
```diff
@@ -0,0 +1,1807 @@
...
+			insert into role_type_permission(role_type_id, permission_id, selectable) values (1, @permission_id , 1);
+			insert into role_type_permission(role_type_id, permission_id, selectable) values (2, @permission_id , 1);
+			insert into role_type_permission(role_type_id, permission_id, selectable) values (3, @permission_id , 1);
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-75">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_FIELD_MAP_OPTIONS';
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1471]: Use add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 5]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-77">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_VIEW_FIELD_MAP';
+             </sqlCheck>
+ 		</preConditions>
+        <comment>View Field Map permission</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1514]: use add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 6]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-78">
+		<preConditions onFail="MARK_RAN">
+ 			<sqlCheck expectedResult="0">
+                 SELECT COUNT(*) FROM PERMISSION P WHERE P.NAME = 'MS_DELETE_FIELD_MAP';
+             </sqlCheck>
+ 		</preConditions>
+        <comment>Delete Field Map permission</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1534]: use add verb in comment

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 7]
```diff
@@ -0,0 +1,1807 @@
...
+ 		</sql>
+ 	</changeSet>
+
+	<changeSet author="cuenyad" id="v23.2.0-81">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="permission" columnName="rank"/>
+			</not>
+		</preConditions>
+        <comment>Adding Rank column</comment>
```
Comment (clarysabel) [inline src/main/resources/liquibase/workbench_changelog/23_2_0.xml:1594]: Adding Rank column to permission table

File: src/main/resources/liquibase/workbench_changelog/23_2_0.xml [hunk 8]
```diff
@@ -0,0 +1,1807 @@
...
+ 		<sql dbms="mysql" splitStatements="true">
+			ALTER TABLE PERMISSION
+				ADD COLUMN rank INT(11) NOT NULL AFTER workbench_sidebar_category_link_id;
+		UPDATE PERMISSION set rank = 1;
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/resources/liquibase/workbench_changelog/23_2_0.xml]: It is because the Rank column is mandatory, and we reorder the rank for Studies only.

**Pull request merged**